server/server.go

```diff
@@ -315,7 +315,7 @@ func (a *ArgoCDServer) watchSettings(ctx context.Context) {
 	prevGitLabSecret := a.settings.WebhookGitLabSecret
 	prevBitBucketUUID := a.settings.WebhookBitbucketUUID
 	var prevCert, prevCertKey string
-	if a.settings.Certificate != nil {
```

Comment:
Argo CD still might generated unnecessary cert if any other setting changes. Instead of updating server `watchSettings` can you please update `InitializeSettings` method to something like this:

https://github.com/argoproj/argo-cd/blob/0f2fe7602735ab12e68d5b7d1dcb70566d4ab4f2/server/server.go#L166

```
settings, err := settingsMgr.InitializeSettings(opts. Insecure)
```

https://github.com/argoproj/argo-cd/blob/6adae305ca0679c79360e11d301583ef36f193c8/util/settings/settings.go#L839

```
if cdSettings.Certificate == nil &&  !insecure {
```